android/core/src/main/java/com/stadiamaps/ferrostar/core/annotation/AnnotationPublisher.kt

```diff
@@ -0,0 +1,7 @@
+package com.stadiamaps.ferrostar.core.annotation
+
+import com.stadiamaps.ferrostar.core.NavigationState
+
+interface AnnotationPublisher<T> {
+  fun map(state: NavigationState): AnnotationWrapper<T>
```

Comment:
I'm totally fine with this pattern because we can use `AnnotationPublisher<*>` 👍

The main reason to avoid generics in this case is is to avoid having to add a generic to every view down the chain (which you'd have to do on iOS for a generic.